### PR TITLE
Runtime/yizhe/beagle compatibility

### PR DESCRIPTION
**Pull request by yizow**

### CODE REVIEW

File: runtime/testy/runtime.py [hunk 1]
```diff
@@ -47,11 +48,14 @@ def runtime():
       print("Starting studentCode attempt: %s" % (restartCount,))
       spawnProcess(PROCESS_NAMES.STUDENT_CODE, runStudentCode)
       while True:
-        newBadThing = badThingsQueue.get(block=True)
```
Comment (nikitakit) [inline runtime/testy/runtime.py:51]: This code is duplicated with `runtimeTest`... and this PR only fixes the bugs in the `runtimeTest` version. You should at least keep the two in sync, and preferably de-duplicate the code asap.

File: runtime/testy/runtimeUtil.py [hunk 2]
```diff
@@ -59,7 +59,7 @@ def __new__(cls):
   GET_TIME            = ()
 
 class RUNTIME_CONFIG(Enum):
-  STUDENT_CODE_TIMEOUT        = 3
+  STUDENT_CODE_TIMEOUT        = 1
```
Comment (nikitakit) [inline runtime/testy/runtimeUtil.py]: BTW the name `STUDENT_CODE_TIMEOUT` is too similar to `BAD_EVENTS.STUDENT_CODE_TIMEOUT`. Also, why is `RUNTIME_CONFIG` an Enum? That makes no sense. I strongly recommend cleaning this up in a future PR.

**Review by nikitakit — CHANGES_REQUESTED** [review]

**Comment by cjsanchez** [comment]
I just added the changes made to runtimeTest() to runtime() and tested that everything ran correctly on the beaglebone. Everything looks good.

File: runtime/testy/runtime.py [hunk 3]
```diff
@@ -66,16 +73,23 @@ def runtime():
 def runStudentCode(badThingsQueue, stateQueue, pipe, testName = "", maxIter = 0):
   try:
     import signal
+
+    terminated = False
+    def sigTermHandler(signum, frame):
+      global terminated
```
Comment (bodiddles) [inline runtime/testy/runtime.py]: Shouldn't this be nonlocal as we only want to change it in the parent frame and not global

File: runtime/testy/runtime.py [hunk 4]
```diff
@@ -172,18 +185,29 @@ def runtimeTest():
...
+              if newBadThing.event in restartEvents:
+                break
+            except Exception as e:
+              print(e)
           stateQueue.put([SM_COMMANDS.RESET, []])
-          os.kill(allProcesses[PROCESS_NAMES.STUDENT_CODE].pid, signal.SIGKILL)
+          student_process = allProcesses[PROCESS_NAMES.STUDENT_CODE]
+          student_process.terminate()
+          time.sleep(.01) # Give the OS a chance to terminate the other process
+          if student_process.is_alive():
```
Comment (bodiddles) [inline runtime/testy/runtime.py]: This code is a repeat of above, we should just have a function that handles this, even if it is not ideal.

**Review by bodiddles — CHANGES_REQUESTED** [review]

File: runtime/testy/runtime.py (hunk 4 above)
Comment (cjsanchez) [reply runtime/testy/runtime.py]: fixed

**Review by nikitakit — APPROVED** [review]
LGTM

**Comment by yizow** [comment]
Force pushed after rebasing. Nothing was changed; all tests still pass.

**Review by bodiddles — APPROVED** [review]

**Pull request merged**